fhir-server/src/main/java/au/csiro/pathling/fhirpath/literal/LiteralPath.java

```diff
@@ -94,9 +92,8 @@ protected LiteralPath(@Nonnull final Dataset<Row> dataset, @Nonnull final Column
       @Nonnull final Type literalValue) {
     this.idColumn = idColumn;
     this.literalValue = literalValue;
-    final DatasetWithColumn datasetWithColumn = createColumn(dataset, buildValueColumn());
-    this.dataset = datasetWithColumn.getDataset();
-    this.valueColumn = datasetWithColumn.getColumn();
+    this.dataset = dataset;
+    this.valueColumn = buildValueColumn();
```

Comment:
This looks like a small change but it may have some unexpected consequences.
For example if we do something like:
``` 1 combine 2 ``` I think results will be [1,1] instead of [1,2] because the value expression from the first literal will be become the value for the combination.
Maybe we should have a way to 'materialize' the literal (or any FhirPath) so that it all columns are in the dataset and all column references that we use (valueColumn) etc, are attribute references rather than expressions.

Reply:
Fixed in f6fe23340eb56c034508b9d5fe5e7bc87a1bc9aa.